pkg/api/types.go

```diff
@@ -898,12 +898,22 @@ func (p *Properties) GetNSGName() string {
 
 // GetPrimaryAvailabilitySetName returns the name of the primary availability set of the cluster
 func (p *Properties) GetPrimaryAvailabilitySetName() string {
```

Comment:
@xizhamsft this method is called to populate AKS metadata for the cluster record. It doesn't make sense that we'd have a static value for `PrimaryAvailabilitySetName` when there is no AvailabilitySet in the cluster. Does changing this method to return an empty string have any breaking effects on the AKS VMSS scenario?